test/FGS.Extensions.Hosting.Middleware.DelayedStart.Tests/DelayedStartHostBuilderExtensionsTests.cs

```diff
@@ -0,0 +1,121 @@
+using System;
+using System.Threading.Tasks;
+using FGS.Tests.Support.Extensions.Hosting.Middleware.Mocking;
+using FGS.Tests.Support.TestCategories;
+using FGS.Tests.Support;
+using FluentAssertions;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using NUnit.Framework;
+using System.Threading;
+
+namespace FGS.Extensions.Hosting.Middleware.DelayedStart.Tests
+{
+    [Unit]
+    [TestFixture]
+    public class DelayedStartHostBuilderExtensionsTests
+    {
+        private IHostBuilder _hostBuilder;
+        private AutoFixture.Fixture _fixture;
+
+        [SetUp]
+        public void SetUp()
+        {
+            HostSpy.Reset();
+            _fixture = new AutoFixture.Fixture();
+            _hostBuilder = CreateHostBuilder();
+        }
+
+        [Test]
+        public async Task WithDelayedStart_GivenDelay_Delays()
+        {
+            var delay = Given_DelayedStart();
+            var maximumDuration = delay + TimeSpan.FromSeconds(1);
```

Comment:
Could this calculation be done inside `BuildAndRunForDurationAsync`? It seems like it is more related to how that method is implemented than what is being tested in these tests.

Reply:
✅